server.R

```diff
@@ -1,385 +1,392 @@
+      parameters_Reform <- parameters_from_df(DF, parameters_column = 3)
+    } else {
+      parameters_SQ  <-  parameters_from_file(req(input$parameters_SQ)$datapath)
+      parameters_Reform <- parameters_from_file(req(input$parameters_Reform)$datapath)
+    }
+    
+    return(list(
+      parameters_SQ = parameters_SQ,
+      parameters_Reform = parameters_Reform
+    ))
+  })
+  
+  
+  # calculate incomes 
+  calculate_income <- reactive({
+    parameters = req(reload_data())
+    
+    # convert inputs
+    MAX_WAGE <- input$max_hours*input$wage1_hourly
+    children <- convert_ages(input$Children_ages)
+    
+    if (input$Acc_type == "Renting"){
+      AS_Accommodation_Rent <- TRUE
+    } else {
+      AS_Accommodation_Rent <- FALSE
+    }
+    
+    if (input$Partnered == 1){
+      partner_wages <- input$gross_wage2*input$hours2
+      partner_hours <- input$hours2
+    } else {
+      partner_wages <- 0
+      partner_hours <- 0
+    }
+    
+    # Create helper emtr function using current inputs
+    hot_emtr <- function(params) {
+      emtr_df <- emtr(
+        # System parameters
+        params,
+        # Family parameters
+        input$Partnered, input$wage1_hourly, children, partner_wages, partner_hours,
+        input$AS_Accommodation_Costs, AS_Accommodation_Rent, as.numeric(input$AS_Area),
+        pov_thresholds = input$pov_thresholds,bhc_median = input$bhc_median,
+        ahc_median = input$ahc_median,
+        # Presentation parameters
+        max_wage = MAX_WAGE, steps_per_dollar = 1L, weeks_in_year = WEEKS_IN_YEAR, 
+        MFTC_WEP_scaling = as.numeric(input$MFTC_WEP_scaling)
+      )
+      return(emtr_df)
+    }
+    
+    X_SQ <- hot_emtr(parameters$parameters_SQ)
+    X_Reform <- hot_emtr(parameters$parameters_Reform)
+    
+
+    
+    # MFTC is meant to make families always better off being off-benefit than staying
+    # on a benefit. We let the user set whether the family stays on benefit
+    # or gets IWTC when they work, with the parameter input$WFFBEN_SQ. This can be:
+    # "Max" - choose the option which maximises the family income.
+    # "WFF" - go off the benefit while working, and get IWTC + MFTC.
+    # "Benefit" - stay on the benefit while working, and never get IWTC or MFTC
+    #             (benefit abates away as earned income increases).
+    # Note that these are only applicable when beneficiaries are ineligible for IWTC;
+    # MFTC eligibility currently depends on IWTC eligibility in the `emtr` function.
+    
+    if (input$WFFBEN_SQ != "WFF") {
+      SQ_params_with_no_IWTC <- remove_IWTC_from_params(parameters$parameters_SQ)
+      X_SQ_without_IWTC <- hot_emtr(SQ_params_with_no_IWTC)
+      
+      if (input$WFFBEN_SQ == "Max") {
+        # Choose which of benefit or IWTC gives max net income
+        X_SQ <- choose_IWTC_or_benefit(X_SQ, X_SQ_without_IWTC)
+        X_SQ[, EMTR := 1 - 1L*(shift(Net_Income,1L,type="lead")-Net_Income)]
+        X_SQ[, EMTR := zoo::na.locf(EMTR)]

```

Comment:
Suggest you move this into the "choose_IWTC_or_benefit" function, same with line 153-154

Reply:
I agree with Luke that you could move this into the function.

Reply:
Will update this